### PR TITLE
Bump Sentry JavaScript 7.3.1

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
Tests fail due to:

> FAIL test/tracing/stalltracking.test.ts (6.517s)
>   ● StallTracking › Stall tracking detects a JS stall
>     TypeError: hub$1.getClient is not a function
>       52 |     setTimeout(() => {
>       53 |       stallTracking.onTransactionFinish(transaction);
>     > 54 |       transaction.finish();
>          |                   ^
>       55 | 
>       56 |       const measurements = getLastEvent()?.measurements;
>       57 | 
>       at Transaction._getBaggageWithSentryValues (node_modules/@sentry/tracing/cjs/span.js:345:24)
>       at Transaction.getBaggage (node_modules/@sentry/tracing/cjs/span.js:304:16)
>       at Transaction.finish (node_modules/@sentry/tracing/cjs/transaction.js:[11](https://github.com/getsentry/sentry-react-native/runs/7000242777?check_suite_focus=true#step:6:12)9:51)
>       at Timeout._onTimeout (test/tracing/stalltracking.test.ts:54:19)
>       at listOnTimeout (node:internal/timers:559:[17](https://github.com/getsentry/sentry-react-native/runs/7000242777?check_suite_focus=true#step:6:18))
>       at processTimers (node:internal/timers:502:7)

**Comment by marandaneto** [comment]
were there any breaking changes related to it?

**Comment by marandaneto** [comment]
> 

Yes, exactly that line of the code, wondering if it's something that JS SDK has to fix or should I should mock it in my tests.

**Comment by Lms24** [comment]
Ahh I think I know what's going on here and I believe mocking `getClient` is fine: Because you're mocking `getCurrentHub` [here](https://github.com/getsentry/sentry-react-native/commit/ad94e98afd3bb34fdc6a719262963a0a734204c1#diff-b48f288acd8a69d671de8736c03991d19e733b9647d6c55ac6877430970360deL11-R18) to return the above created `hub` object, the `getClient` method was simply missing in the mock. The Transaction we access in  `_getBaggageWithSentryValues`, when started/created, defaults to setting `getCurrentHub` as its hub which is what we end up using (or we directly call `getCurrentHub` if the transaction has no hub).

> wondering if it's something that JS SDK has to fix

I don't think so (if my assumption above is correct). Can't really change this on the JS SDK though because we need the client at this stage to collect the information for the dynamic sampling context (i.e. the stuff we propagate via baggage and send in the envelope header).

**Comment by marandaneto** [comment]
Thanks @Lms24 appreciate it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +19,7 @@
 ## 4.0.0-beta.4
 
 - Bump Sentry Cocoa 7.17.0 ([#2300](https://github.com/getsentry/sentry-react-native/pull/2300))
+
```
Comment (marandaneto) [inline CHANGELOG.md:22]:
```suggestion
```

File: ios/RNSentry.m [hunk 2]
```diff
@@ -273,7 +274,9 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
     if ([NSJSONSerialization isValidJSONObject:envelopeDict]) {
         SentrySdkInfo *sdkInfo = [[SentrySdkInfo alloc] initWithDict:envelopeDict[@"header"]];
         SentryId *eventId = [[SentryId alloc] initWithUUIDString:envelopeDict[@"header"][@"event_id"]];
-        SentryEnvelopeHeader *envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId sdkInfo:sdkInfo traceContext:nil];
+        // forward the baggage header, right now only trace_id and public_key
```
Comment (marandaneto) [inline ios/RNSentry.m:277]: Wait for https://github.com/getsentry/sentry-javascript/pull/5292 to be released.
@Lms24 please ping me once this is released :) thanks.

Comment (Lms24) [reply ios/RNSentry.m:277]: Will do (will finish the PR today, so probably merged tomorrow and released soon afterwards)

Comment (marandaneto) [reply ios/RNSentry.m:277]: @Lms24 so `7.3.0` has all the changes required by https://github.com/getsentry/develop/pull/613 right?

Comment (Lms24) [reply ios/RNSentry.m:277]: @marandaneto almost all of them (which is the reason I was hesitant to tag you). The only thing the SDK is currently still missing is the correct treatment of 3rd party baggage header items on incoming requests (we still propagate them to outgoing requests which we shouldn't). Currrently working on that and we'll release a path the next days.

Comment (marandaneto) [reply ios/RNSentry.m:277]: Thanks, we'll wait for this patch then.

Comment (marandaneto) [reply ios/RNSentry.m:277]: Thanks.

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,6 +43,7 @@ By bumping Sentry Javascript, new breaking changes were introduced, to know more
...
```
Comment (marandaneto) [inline CHANGELOG.md:46]:
```suggestion
```

**Review by Lms24 — APPROVED** [review]

**Pull request merged**